src/hooks/useIsMounted/useIsMounted.stories.mdx

```diff
@@ -0,0 +1,112 @@
+ref object is updated to `false`. This is because the cleanup phase is executed before the
+component is technically unmounted. However, since the logic during cleanup and after unmount is
+often the same, this is not a problem.
+
+The initial `false` state before the component is mounted can be useful to conditionally execute
+some logic or render something. However, this hook will not cause a re-render, so it's best to use
+alternative hooks mentioned below if that is required.
+
+The `false` state during and after the cleanup phase is useful for preventing memory leaks and
+preventing state updates after the component is unmounted. This is often the case when using
+`useEffect` with an async function, and the async function is still running after the component
+is unmounted.
+
+The state is set to `true` using a `queueMicrotask` callback, so it will be updated after all the
+initial `useEffect` callbacks have been executed.
+
+If the component uses a setState directly from a `useMount`, the component will rerender during the
+mount phase, and the isMounted value will still be `false`.
+
+> If code needs to be executed before mount and is more isolated as a side effect, use the
+`useBeforeMount` hook instead.
+
+> If the code result is different before and after mount, and should execute again, use the
+`useIsMountedState` hook instead.
+
+> If code execution is required after the component is **mounted**, use the `useMount` hook.
+
+> If code execution is required after the component is **unmounted**, use the `useUnmount`
+hook.
+
+## Reference
+
+```ts
+function useIsMounted(): RefObject<boolean>;
+```
+
+### Returns
+
+* `RefObject<boolean>` - The current mounting state.
+  - `false` before the component is mounted
+  - `true` after the component is mounted
+  - `false` during the unmount cleanup phase and after the component is unmounted
+
+## Usage
+
+```ts
+const isMounted = useIsMounted();
+console.log(isMounted.current); // false
+```
+
+Use after unmount in async functions, to prevent memory leaks and state updates after unmount.
+
+```tsx
+function DemoComponent() {
+  const isMounted = useIsMounted();
+
+  useMount(() => {
+    fetchSomething().then(() => {
+      // check if the component is still mounted after the async operation
+      if (isMounted.current) {
+        // do something with the result
+      }
+    });
+  })
+
+  return (
+    <div>
+    </div>
+  );
+}
+```
+
+Use in `useEffect` cleanup, for some very specific scenarios.
+
+```tsx
+import { useEffect } from 'react';
+function DemoComponent() {
+  const isMounted = useIsMounted();
+
+  useEffect(() => {
+    // do something
+
+    return () => {
+      if (isMounted.current) {
+        // this is executed every time someChangingValue changes
+      } else {
+        // this is executed only once, when the component is unmounted
+      }
+    }
+  }, [someChangingValue]);
```

Comment:
`someChangingValue` isn't defined anywhere, maybe define it as a prop on `DemoComponent` to avoid confusion in the docs.